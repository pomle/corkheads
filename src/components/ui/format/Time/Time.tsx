import React from "react";
import { Moment } from "moment";

const FORMAT = "HH:mm";

interface TimeProps {
  date: Moment;
}

const Time: React.FC<TimeProps> = ({ date }) => {
  if (!date.isValid()) {
    return <>--:--</>;
  }

  return <>{date.format(FORMAT)}</>;
};

export default Time;
