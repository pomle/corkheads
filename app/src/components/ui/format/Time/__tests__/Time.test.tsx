import React from "react";
/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import renderer from "react-test-renderer";
import moment from "moment";
import Time from "../Time";

describe("Time component", () => {
  const VALID_MOMENT = moment("2019-09-12T12:42:13.423Z");
  const INVALID_MOMENT = moment.invalid();

  it("shows the time of day for the date when valid", () => {
    const component = renderer.create(<Time date={VALID_MOMENT} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("shows blank time when date invalid", () => {
    const component = renderer.create(<Time date={INVALID_MOMENT} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
