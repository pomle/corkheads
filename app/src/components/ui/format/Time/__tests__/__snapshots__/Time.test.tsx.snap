// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time component shows blank time when date invalid 1`] = `"--:--"`;

exports[`Time component shows the time of day for the date when valid 1`] = `"12:42"`;
